src/interfaces.rs

```diff
@@ -271,6 +264,32 @@ impl FromIterator<(String, Interface)> for Interfaces {
     }
 }
 
+struct Introspection<'a> {
+    interfaces: &'a HashMap<String, Interface>,
+}
+
+impl<'a> Display for Introspection<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut iter = self.interfaces.iter();
+
+        if let Some((name, interface)) = iter.next() {
+            let major = interface.version_major();
+            let minor = interface.version_minor();
+            write!(f, "{}:{}:{}", name, major, minor)?;
+        } else {
+            return Ok(());
+        }
+
+        for (name, interface) in iter {
+            let major = interface.version_major();
+            let minor = interface.version_minor();
+            write!(f, ";{}:{}:{}", name, major, minor)?;
+        }
```

Comment:
What do yopu think of the following?
```suggestion
        let Some((name, interface)) = iter.next() else {
            return Ok(());
        }

        let mut major = interface.version_major();
        let mut minor = interface.version_minor();
        write!(f, "{}:{}:{}", name, major, minor)?;

        for (name, interface) in iter {
            major = interface.version_major();
            minor = interface.version_minor();
            write!(f, ";{}:{}:{}", name, major, minor)?;
        }
```

Reply:
This is the forward port from 1.59 this syntax was not supported in that rust version

Reply:
I will add a commit to refactor this

Reply:
Done